Bio/AlignIO/MafIO.py

```diff
@@ -0,0 +1,581 @@
+
+        # for now, use ._annotations private property, but restrict keys to those
+        # specifically supported by the MAF format, according to spec
+        try:
+            anno = " ".join(["%s=%s" % (x, y) for x, y in alignment._annotations.items() if x in ("score", "pass")])
+        except AttributeError:
+            anno = "score=0.00"
+
+        self.handle.write("a %s\n" % (anno,))
+
+        recs_out = 0
+
+        for record in alignment:
+            self._write_record(record)
+
+            recs_out += 1
+
+        self.handle.write("\n")
+
+        return recs_out
+
+
+def MafIterator(handle, seq_count=None, alphabet=single_letter_alphabet):
+    """
+    Iterates over lines in a MAF file-like object (handle), yielding
+    MultipleSeqAlignment objects. SeqRecord IDs generally correspond to
+    species names
+    """
+
+    in_a_bundle = False
+
+    annotations = []
+    records = []
+
+    while True:
+        # allows parsing of the last bundle without duplicating code
+        try:
+            line = next(handle)
+        except StopIteration:
+            line = ""
+
+        if in_a_bundle:
+            if line.startswith("s"):
+                # add a SeqRecord to the bundle
+                line_split = line.strip().split()
+
+                if len(line_split) != 7:
+                    raise ValueError("Error parsing alignment - 's' line must have 7 fields")
+
+                # convert MAF-style +/- strand to biopython-type 1/-1
+                if line_split[4] == "+":
+                    strand = 1
+                elif line_split[4] == "-":
+                    strand = -1
+                else:
+                    strand = 1
+
+                # s (literal), src (ID), start, size, strand, srcSize, text (sequence)
+                anno = {"start": int(line_split[2]),
+                        "size": int(line_split[3]),
+                        "strand": strand,
+                        "srcSize": int(line_split[5])}
+
+                sequence = line_split[6]
+
+                # interpret a dot/period to mean same the first sequence
+                if "." in sequence:
+                    if not records:
+                        raise ValueError("Found dot/period in first sequence of alignment")
+
+                    ref = str(records[0].seq)
+                    new = []
+
+                    for (s, r) in zip(sequence, ref):
+                        new.append(r if s == "." else s)
+
+                    sequence = "".join(new)
+
+                records.append(SeqRecord(Seq(sequence, alphabet),
+                               id=line_split[1],
+                               name=line_split[1],
+                               description="",
+                               annotations=anno))
+            elif line.startswith("e") or \
+                 line.startswith("i") or \
+                 line.startswith("q"):
+                # not implemented
```

Comment:
Good question, or as a middle ground:

``` python
import warnings
from Bio import BiopythonParserWarning
...
warnings.warn("MAF line type %r not implemented" % line[0], BiopythonParserWarning)
```

If the extra lines are fairly minor then ignoring them silently is probably OK.